### ISSUE TITLE
minio web console duplicate prefix

### ISSUE THREAD

**Issue by fdefilippo** [minio/minio#10881]
With the latest release of minio, accessing via web gui displays double prefixes:

![chrome_Qj6zDgUofj](https://user-images.githubusercontent.com/16714986/98912608-9eb94a80-24c6-11eb-86b5-87f6000840d0.png)

$ mc ls --versions local/francesco
[2020-10-31 06:22:37 CET]     0B null v1 PUT windows/
[2020-11-12 09:05:07 CET]     0B test/
[2020-11-12 09:05:07 CET]     0B windows/

$ mc ls local/francesco/
[2020-10-31 06:22:37 CET]     0B windows/
[2020-11-12 09:12:00 CET]     0B test/
[2020-11-12 09:12:00 CET]     0B windows/

what does it mean "null v1 PUT" ?

with previous (RELEASE.2020-10-28T08-16-50Z) version of minio:

$ mc ls local/francesco/
[2020-11-12 08:51:24 CET]     0B windows/
[2020-11-12 09:05:07 CET]     0B test/

## Expected Behavior

that shows the correct list 

## Steps to Reproduce (for bugs)

updagrade minio to latest version

## Your Environment

$ minio -version
minio version RELEASE.2020-11-10T21-02-24Z

ubuntu 18.x and centos 8.x

**Comment by vadmeste** [comment]
Yeah, this looks like a MinIO issue:

```
aws --profile play --endpoint-url https://play.min.io s3api list-objects --bucket vadmeste-testbucket --delimiter '/'
{
    "Contents": [
        {
            "Key": "windows/",
            "LastModified": "2020-11-12T11:15:33.492Z",
            "ETag": "\"d41d8cd98f00b204e9800998ecf8427e\"",
            "Size": 0,
            "StorageClass": "STANDARD",
            "Owner": {
                "DisplayName": "",
                "ID": "02d6176db174dc93cb1b899f7c6078f08654445fe8cf1b6ce98d8855f66bdbf4"
            }
        }
    ],
    "CommonPrefixes": [
        {
            "Prefix": "windows/"
        }
    ]
}
```

It should show this instead
```
{
    "CommonPrefixes": [
        {
            "Prefix": "windows/"
        }
    ]
}

```

**Comment by klauspost** [comment]
@vadmeste Can you see if this is because of the directory hack?

**Comment by harshavardhana** [comment]
No it is because we should skip with Delimiter an actual object entry.

**Comment by harshavardhana** [comment]
```diff
git diff
diff --git a/cmd/metacache-entries.go b/cmd/metacache-entries.go
index 39e31f18e..7e2a881ab 100644
--- a/cmd/metacache-entries.go
+++ b/cmd/metacache-entries.go
@@ -357,6 +357,7 @@ func (m *metaCacheEntriesSorted) fileInfos(bucket, prefix, delimiter string) (ob
        objects = make([]ObjectInfo, 0, m.len())
        prevPrefix := ""
        for _, entry := range m.o {
+               entry.name = decodeDirObject(entry.name)
                if entry.isObject() {
                        if delimiter != "" {
                                idx := strings.Index(strings.TrimPrefix(entry.name, prefix), delimiter)
```

This change should fix the problem

**Comment by i0x71** [comment]
Can buckets which have duplicate entries, be safely used ?

**Comment by harshavardhana** [comment]
> Can buckets which have duplicate entries, be safely used ?

yes they can be, it is just a cosmetic bug @i0x71

**Comment by harshavardhana** [comment]
Fixed by #10886
